qa/tasks/cephfs/test_snap_schedules.py

```diff
@@ -0,0 +1,185 @@
+            self.mount_a.run_shell(["rmdir", snapshot_path])
+
+    def test_snap_schedule(self):
+        self.mount_a.run_shell(["mkdir", "-p", TestSnapSchedules.TEST_DIRECTORY])
+
+        # set a schedule on the dir
+        self._fs_snap_schedule_cmd("add", TestSnapSchedules.TEST_DIRECTORY, '2m')
+        exec_time = time.time()
+
+        # verify snapshot schedule
+        self._verify_schedule(TestSnapSchedules.TEST_DIRECTORY, [['2m', '']])
+
+        # wait for the snapshot
+        self._verify_snapshot(TestSnapSchedules.TEST_DIRECTORY, '2m', exec_time)
+
+        # remove snapshot schedule
+        self._fs_snap_schedule_cmd("remove", TestSnapSchedules.TEST_DIRECTORY)
+        try:
+            self._fs_snap_schedule_cmd("list", TestSnapSchedules.TEST_DIRECTORY)
+        except CommandFailedError as ce:
+            if ce.exitstatus != errno.EPERM:
+                raise RuntimeError("incorrect errno when listing a non-existing snap schedule")
+
+        # remove all scheduled snapshots
+        self._remove_snapshots(TestSnapSchedules.TEST_DIRECTORY)
+
+        self.mount_a.run_shell(["rmdir", TestSnapSchedules.TEST_DIRECTORY])
+
+    def test_multi_snap_schedule(self):
+        self.mount_a.run_shell(["mkdir", "-p", TestSnapSchedules.TEST_DIRECTORY])
+
+        # set schedules on the dir
+        self._fs_snap_schedule_cmd("add", TestSnapSchedules.TEST_DIRECTORY, '2m')
+        self._fs_snap_schedule_cmd("add", TestSnapSchedules.TEST_DIRECTORY, '3m')
+        exec_time = time.time()
+
+        # verify snapshot schedule
+        self._verify_schedule(TestSnapSchedules.TEST_DIRECTORY, [['2m', ''], ['3m', '']])
+
+        # wait for the snapshot
+        self._verify_snapshot(TestSnapSchedules.TEST_DIRECTORY, '2m', exec_time)
+        self._verify_snapshot(TestSnapSchedules.TEST_DIRECTORY, '3m', exec_time, existing_snap_count=1)
+
+        # remove snapshot schedule
+        self._fs_snap_schedule_cmd("remove", TestSnapSchedules.TEST_DIRECTORY)
+
+        # remove all scheduled snapshots
+        self._remove_snapshots(TestSnapSchedules.TEST_DIRECTORY)
+
+        self.mount_a.run_shell(["rmdir", TestSnapSchedules.TEST_DIRECTORY])
+
+    def test_snap_schedule_on_multiple_directory(self):
+        test_dir0 = "{0}.0".format(TestSnapSchedules.TEST_DIRECTORY)
+        test_dir1 = "{0}.1".format(TestSnapSchedules.TEST_DIRECTORY)
+
+        self.mount_a.run_shell(["mkdir", "-p", test_dir0])
+        self.mount_a.run_shell(["mkdir", "-p", test_dir1])
+        exec_time = time.time()
+
+        # set schedule on the dirs
+        self._fs_snap_schedule_cmd("add", test_dir0, '1m')
+        self._fs_snap_schedule_cmd("add", test_dir1, '2m')
+
+        # veridy snapshot schedules
+        self._verify_schedule(test_dir0, [['1m', '']])
+        self._verify_schedule(test_dir1, [['2m', '']])
+
+        # wait for the snapshot
+        self._verify_snapshot(TestSnapSchedules.TEST_DIRECTORY, '2m', exec_time)
```

Comment:
self._verify_snapshot(test_dir0)?

Reply:
yeh -- I fixed this locally.

Wanted to get the updated PR for @jan--f to reproduce the isssue.